### PR TITLE
feat: add `platformProxy` option to configure `getPlatformProxy`

### PR DESCRIPTION
**Pull request by vnphanquang**
> [!NOTE]
> I've run `check` & `lint` but not all `test` scripts because I don't see any specific tests in the two packages' `package.json`, and some of tests fail on my machine (`chromium-dev` related) but I don't think they are related to the change here.

## Context

Today, `adapter-cloudflare` and `adapter-cloudflare-worker` use Wrangler API's [getPlatformProxy](https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy) for `platform` emulation during `dev` & `preview`, but not yet allow passing configuration parameter to `getPlatformProxy`. This PR adds a `platformProxy` adapter option to do just that.

## Use case

My specific use case is to set `persist.path` to a custom directory (equivalent to the [--persist-to](https://developers.cloudflare.com/d1/configuration/local-development/#persist-data) flag when using wrangler CLI). This allows me to set up two SvelteKit projects in a monorepo to use the same local sqlite database during dev & preview.

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: .changeset/tough-dolphins-kick.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@sveltejs/adapter-cloudflare-workers": minor
+"@sveltejs/adapter-cloudflare": minor
+---
+
+support platform emulation configuration via the `platformProxy` adapter option
```
Comment (benmccann) [inline .changeset/tough-dolphins-kick.md:6]:
```suggestion
feat: support platform emulation configuration via the `platformProxy` adapter option
```

File: documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md [hunk 2]
```diff
@@ -121,6 +121,8 @@ Cloudflare Workers specific values in the `platform` property are emulated durin
 
 For testing the build, you should use [wrangler](https://developers.cloudflare.com/workers/cli-wrangler) **version 3**. Once you have built your site, run `wrangler dev`.
 
+Under the hood, `adapter-cloudflare` uses [getPlatformProxy](https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy) from Wrangler API for `platform` emulation. Further configuration can be passed to `getPlatformProxy` via the `platformProxy` option in your adapter config.
```
Comment (benmccann) [inline documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md:124]:
```suggestion
Under the hood, `adapter-cloudflare-workers` uses [getPlatformProxy](https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy) from Wrangler API for `platform` emulation. Further configuration can be passed to `getPlatformProxy` via the `platformProxy` option in your adapter config.
```

File: .changeset/tough-dolphins-kick.md (hunk 1 above)
Comment (vnphanquang) [reply .changeset/tough-dolphins-kick.md]: Thanks for the feedback. Amended to previous commit to keep it clean.

File: documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md (hunk 2 above)
Comment (vnphanquang) [reply documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md]: Thanks for the feedback. Amended to previous commit to keep it clean.

**Comment by hrueger** [comment]
Also very interested in this 👍 Is there anything that can be done to move this forward like testing or so?

My use case is the following (I believe it is not that uncommon):
I have multiple sveltekit apps in a monorepo. They all share the same bindings, so the wrangler data folders need to be at the root of the monorepo, not in each single package. This can be configured in the options of `getPlatformProxy()`, therefore this PR would fix that issue 🎉

**Comment by AskingQuestions** [comment]
> Also very interested in this 👍 Is there anything that can be done to move this forward?
> 
> My use case is the following (I believe it is not that uncommon): I have multiple sveltekit apps in a monorepo. They all share the same bindings, so the wrangler data folders need to be at the root of the monorepo, not in each single package. This can be configured in the options of `getPlatformProxy()`, therefore this PR would fix that issue 🎉

+1 to this. Same exact use case over here. I'm currently using a patched version of the adapter-cloudflare package to workaround this.

File: packages/adapter-cloudflare-workers/index.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ const compatible_node_modules = [
 ];
 
 /** @type {import('./index.js').default} */
-export default function ({ config = 'wrangler.toml' } = {}) {
+export default function ({ config = 'wrangler.toml', platformProxy } = {}) {
```
Comment (eltigerchino) [inline packages/adapter-cloudflare-workers/index.js:35]:
```suggestion
export default function ({ config = 'wrangler.toml', platformProxy = undefined } = {}) {
```

File: packages/adapter-cloudflare-workers/index.d.ts [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
 import './ambient.js';
+import { GetPlatformProxyOptions } from 'wrangler';
 
 export default function plugin(options?: AdapterOptions): Adapter;
 
 export interface AdapterOptions {
 	config?: string;
+	/**
+	 * Config object passed to {@link https://developers.cloudflare.com/workers/wrangler/api/#getplatformproxy | getPlatformProxy}
+	 * during development & preview
```
Comment (eltigerchino) [inline packages/adapter-cloudflare-workers/index.d.ts:11]:
```suggestion
	 * during development and preview.
```

**Review by eltigerchino — APPROVED** [review]

File: packages/adapter-cloudflare-workers/index.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ const compatible_node_modules = [
 ];
 
 /** @type {import('./index.js').default} */
-export default function ({ config = 'wrangler.toml' } = {}) {
+export default function ({ config = 'wrangler.toml', platformProxy = undefined } = {}) {
```
Comment (eltigerchino) [inline packages/adapter-cloudflare-workers/index.js:35]:
```suggestion
export default function ({ config = 'wrangler.toml', platformProxy = {} } = {}) {
```

File: documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md [hunk 6]
```diff
@@ -63,19 +78,7 @@ wrangler deploy
...
-/// file: svelte.config.js
-import adapter from '@sveltejs/adapter-cloudflare-workers';
-
-export default {
-	kit: {
-		adapter: adapter({ config: '<your-wrangler-name>.toml' })
-	}
-};
-```
+If you would like to use a config file other than `wrangler.toml` you can specify so using the [config option](#config).
```
Comment (eltigerchino) [inline documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md:81]:
```suggestion
If you would like to use a config file other than `wrangler.toml` you can specify so using the [`config` option](#config).
```

File: documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md [hunk 7]
```diff
@@ -90,7 +99,7 @@ export {};
 
 ### Testing Locally
 
-Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development.
+Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#platformProxy) to change your preferences for the bindings.
```
Comment (eltigerchino) [inline documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md:102]:
```suggestion
Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#options-platformProxy) to change your preferences for the bindings.
```

File: documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md [hunk 8]
```diff
@@ -117,7 +120,7 @@ export {};
 
 ### Testing Locally
 
-Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development.
+Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#platformProxy) to change your preferences for the bindings.
```
Comment (eltigerchino) [inline documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md:123]:
```suggestion
Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#options-platformProxy) to change your preferences for the bindings.
```

File: documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md [hunk 9]
```diff
@@ -63,19 +78,7 @@ wrangler deploy
...
-/// file: svelte.config.js
-import adapter from '@sveltejs/adapter-cloudflare-workers';
-
-export default {
-	kit: {
-		adapter: adapter({ config: '<your-wrangler-name>.toml' })
-	}
-};
-```
+If you would like to use a config file other than `wrangler.toml` you can specify so using the [`config` option](#config).
```
Comment (eltigerchino) [inline documentation/docs/25-build-and-deploy/70-adapter-cloudflare-workers.md:81]:
```suggestion
If you would like to use a config file other than `wrangler.toml` you can specify so using the [`config` option](#options-config).
```

File: documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md [hunk 10]
```diff
@@ -90,7 +99,7 @@ export {};
 
 ### Testing Locally
 
-Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development.
+Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#options-platformProxy) to change your preferences for the bindings.
```
Comment (eltigerchino) [inline documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md:102]:
```suggestion
Cloudflare Workers specific values in the `platform` property are emulated during dev and preview modes. Local [bindings](https://developers.cloudflare.com/workers/wrangler/configuration/#bindings) are created based on the configuration in your `wrangler.toml` file and are used to populate `platform.env` during development and preview. Use the adapter config [`platformProxy` option](#options-platformproxy) to change your preferences for the bindings.
```

**Comment by eltigerchino** [comment]
Thank you!

**Comment by vnphanquang** [comment]
Thank you team :tada:

**Comment by hrueger** [comment]
Thanks for merging this, could we maybe get a new release? ;-)

**Pull request merged**